plutarch-test/plutarch-test.cabal

```diff
@@ -134,10 +150,6 @@ executable plutarch-test
     Plutarch.RecursionSpec
     Plutarch.ScriptsSpec
     Plutarch.StringSpec
-    Plutarch.Test
-    Plutarch.Test.Benchmark
-    Plutarch.Test.Golden
-    Plutarch.Test.ListSyntax
     Plutarch.TraceSpec
```

Comment:
@MangoIV These test library modules are originally replicated in the executable stanza for a reason. I use `ghcid` to run the tests, and with this replication - changing the library modules automatically recompiles/re-runs the tests. Otherwise, you have to restart ghcid itself every time one of those test library modules change.

It is just 4 modules, so I figured it was worth doing this at the slight expense of longer CI times.  I'd keep it as is, and extend the replication to plutarch-extra-test as well.